mlflow/java/src/main/java/org/mlflow/sagemaker/LeapFrameSchema.java

```diff
@@ -0,0 +1,92 @@
+package org.mlflow.sagemaker;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.mlflow.utils.SerializationUtils;
+
+/** Representation of the dataframe schema that an {@link MLeapPredictor} expects inputs to have */
+@JsonIgnoreProperties(ignoreUnknown = true)
+class LeapFrameSchema {
+  private static final String LEAP_FRAME_KEY_ROWS = "rows";
+  private static final String LEAP_FRAME_KEY_SCHEMA = "schema";
+
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  static class SchemaField {
+    @JsonProperty("name")
+    private String name;
+  }
+
+  @JsonProperty("fields")
+  private List<SchemaField> fields;
+
+  private String text;
+  private List<String> orderedFieldNames;
+
+  /**
+   * Loads a leap frame schema from a JSON-formatted file
+   *
+   * @param filePath The path to the JSON-formatted schema file
+   */
+  protected static LeapFrameSchema fromPath(String filePath) throws IOException {
+    LeapFrameSchema newSchema =
+        SerializationUtils.parseJsonFromFile(filePath, LeapFrameSchema.class);
+    String schemaText = new String(Files.readAllBytes(Paths.get(filePath)));
+    newSchema.setText(schemaText);
```

Comment:
Can you document when IOException will be thrown? What Exception is thrown if the schema is invalid?

Reply:
Added documentation indicating that:

1. `IOException` is thrown when the specified file cannot be loaded (i.e. `new File(filePath)` fails)
2. `InvalidSchemaException` is thrown if the file's schema JSON cannot be parsed